pkg/scaler/prober.go

```diff
@@ -508,6 +517,9 @@ func (p *prober) scaleTo(parentContext context.Context, msg string, replicas int
 				if !checkFn(specReplicas, replicas) {
 					klog.V(4).Infof("%s: skipped because desired=%d and current=%d", prefix, replicas, specReplicas)
 					continue
+				} else if hasIgnoreScalingAnnotation(d) {
+					klog.V(4).Infof("%s: skipped because ignore scaling annotation present on deployment", prefix)
```

Comment:
This will be applicable for both scaleUp and scaleDown operations. 
So what happens if - 
1. DWD brings down the MCM or other deployment as a result of the external probe failing and then someone applies this annotation to that resource. 
2. What if you wanted to only do it while scaling down but when scaling up or vice-versa.

Reply:
yes this annotation doesn't restrict to scale-up ignoring only. The only `someone` to be expected to apply it IT. The IT after being done will scale up the deployment and remove the annotation, so its won't happen that MCM/CA won't come up.

Reply:
I was thinking for your integration test you would like to ensure that DWD doesn't scale up CA immediately if the annotation is applied. Why don't we deploy the config for DWD probe with the following 2 options -- 

1.  Remove `cluster-autoscalar` from dependentScales  in which case its ignored.  (this can be then restored to the default configs post tests. 

```
  dep-config.yaml: |
    namespace: ""
    probes:
    - dependantScales:
      - replicas: null
        scaleRef:
          apiVersion: apps/v1
          kind: Deployment
          name: kube-controller-manager
        scaleUpDelaySeconds: 120
      - replicas: null
        scaleRef:
          apiVersion: apps/v1
          kind: Deployment
          name: machine-controller-manager
        scaleRefDependsOn:
        - apiVersion: apps/v1
          kind: Deployment
          name: kube-controller-manager
        scaleUpDelaySeconds: 60
```

or 
2. you can a scaleUpDelay for CA to ensure that it is after your tests are passed. (this can be a permanent config for a given landscape as eventually CA will be scaled back and its not production critical. 
```
  dep-config.yaml: |
    namespace: ""
    probes:
    - dependantScales:
      - replicas: null
        scaleRef:
          apiVersion: apps/v1
          kind: Deployment
          name: kube-controller-manager
        scaleUpDelaySeconds: 120
      - replicas: null
        scaleRef:
          apiVersion: apps/v1
          kind: Deployment
          name: machine-controller-manager
        scaleRefDependsOn:
        - apiVersion: apps/v1
          kind: Deployment
          name: kube-controller-manager
        scaleUpDelaySeconds: 60
      - replicas: null
        scaleRef:
          apiVersion: apps/v1
          kind: Deployment
          name: cluster-autoscaler
        scaleUpDelaySeconds: 300
        scaleRefDependsOn:
        - apiVersion: apps/v1
          kind: Deployment
          name: machine-controller-manager
```

cc @unmarshall

Reply:
I'm fine with keep  this as is for now. 
We can rethink on the solution when we refactor to see if there can be a better generic approach to handle such scenarios. 

/lgtm